### PR TITLE
Paper - Heidi Zhang - Swap Meet 

### PR DESCRIPTION
**Pull request by qingyuanhz1**

### CODE REVIEW

File: swap_meet/clothing.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from swap_meet.item import Item
+
+class Clothing(Item):
```
Comment (CheezItMan) [inline swap_meet/clothing.py:3]: 👍 Very clean child class

File: swap_meet/decor.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from swap_meet.item import Item
+
+class Decor(Item):
```
Comment (CheezItMan) [inline swap_meet/decor.py:3]: 👍 One question, why default age to 1000?  Antiques?

File: swap_meet/electronics.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from swap_meet.item import Item
+
+class Electronics(Item):
```
Comment (CheezItMan) [inline swap_meet/electronics.py:3]: 👍

File: swap_meet/item.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+class Item:
```
Comment (CheezItMan) [inline swap_meet/item.py:1]: 👍

File: swap_meet/vendor.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
+from swap_meet.item import Item
+
+class Vendor: 
+    def __init__(self, inventory=None):
+        if inventory == None: 
+            self.inventory = []
```
Comment (CheezItMan) [inline swap_meet/vendor.py:6]: 👍

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        else: 
+            self.inventory = inventory
+
+    def add(self, add_item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:10]: 👍

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        self.inventory.append(add_item)
+        return add_item    
+    
+    def remove(self, rm_item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:14]: 👍

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        return add_item    
+    
+    def remove(self, rm_item): 
+        if rm_item in self.inventory:
+            self.inventory.remove(rm_item)
+            return rm_item
+        else:
+            return rm_item  == False 
+    
+    def get_by_category(self, asked_category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:21]: 👍

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    
+    def swap_items(self, other_vendor, my_item, their_item): 
+        if my_item in self.inventory and their_item in other_vendor.inventory: 
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            return True 
+        else: 
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:37]: The `else` isn't needed.

```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+    def swap_first_item(self, other_vendor):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:39]: 👍 Similar extra else here.

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+    def swap_first_item(self, other_vendor): 
+        if len(self.inventory) and len(other_vendor.inventory) > 0: 
+            self_inventory_1 = self.inventory[0]  
+            other_vendor_invetory_1 = other_vendor.inventory[0]
+            return self.swap_items(other_vendor, self_inventory_1, other_vendor_invetory_1)
+        else: 
+            return False
+    
+    def get_best_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:47]: 👍

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+                initial_condition = item.condition
+                best_item = item
+        
+        return best_item
+    
+    def swap_best_by_category(self, other, my_priority, their_priority):
+        my_best_item = self.get_best_by_category(their_priority)
+        their_best_item = other.get_best_by_category(my_priority)
+        
+        if my_best_item != None and their_best_item != None:
```
Comment (CheezItMan) [inline swap_meet/vendor.py:63]: Since `None` is falsey, you can do this.

```suggestion
        if my_best_item and their_best_item:
```

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+        best_item = None
+
+        for item in item_list: 
+            if item.condition > initial_condition: 
+                initial_condition = item.condition
+                best_item = item
+        
+        return best_item
+    
+    def swap_best_by_category(self, other, my_priority, their_priority):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:59]: 👍 , also an extraneous `else`

File: tests/test_wave_enhancement.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c]
+    )
+
+    newest_item = tai.get_best_by_category("Electronics")
+
+    assert newest_item is None
+
+
+def test_newest_by_category_with_duplicates():
```
Comment (CheezItMan) [inline tests/test_wave_enhancement.py:38]: Good edge-case test

File: tests/test_wave_enhancement.py [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    assert result is True
+    assert len(tai.inventory) is 3
+    assert len(jesse.inventory) is 3
+    assert item_c not in tai.inventory
+    assert item_f in tai.inventory
+    assert item_f not in jesse.inventory
+    assert item_c in jesse.inventory
+
+
+def test_swap_by_newest_no_match_is_false():
```
Comment (CheezItMan) [inline tests/test_wave_enhancement.py:82]: Good pair of edge-case tests.

You should also test when there are items tied for newest.

File: tests/test_wave_enhancement.py [hunk 16]
```diff
@@ -0,0 +1,131 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+
+def test_newest_by_category():
```
Comment (CheezItMan) [inline tests/test_wave_enhancement.py:9]: Nice test, you should also test when there are no items.

**Review by CheezItMan — COMMENTED** [review]
Great work Heidi, you hit the learning goals here.  You even got the optionals working and tests for the optionals.  I like the edge-cases you included as well.  Well done.  I did have some comments, mostly about coding style.  Let me know if you have questions.
